src/cfnlint/maintenance.py

```diff
@@ -21,19 +22,43 @@
 def update_resource_specs():
     """ Update Resource Specs """
 
-    for region, url in SPEC_REGIONS.items():
-        filename = os.path.join(os.path.dirname(cfnlint.__file__),
-                                'data/CloudSpecs/%s.json' % region)
-        LOGGER.debug('Downloading template %s into %s', url, filename)
-        spec = json.loads(get_url_content(url))
+    # Pool() uses cpu count if no number of processors is specified
+    # Pool() only implements the Context Manager protocl from Python3.3 onwards,
+    # so it will fail Python2.7 style linting, as well as throw AttributeError
+    try:
+        # pylint: disable=not-context-manager
+        with multiprocessing.Pool() as pool:
+            pool.starmap(update_resource_spec, SPEC_REGIONS.items())
+    except AttributeError:
 
-        # Patch the files
-        spec = patch_spec(spec, 'all')
-        spec = patch_spec(spec, region)
+        # Do it the long, slow way
+        for region, url in SPEC_REGIONS.items():
+            update_resource_spec(region, url)
 
-        with open(filename, 'w') as f:
-            json.dump(spec, f, indent=2, sort_keys=True, separators=(',', ': '))
+def update_resource_spec(region, url):
+    """ Update a single resource spec """
+    filename = os.path.join(os.path.dirname(cfnlint.__file__), 'data/CloudSpecs/%s.json' % region)
 
+    multiprocessing_logger = multiprocessing.log_to_stderr()
+
+    multiprocessing_logger.debug('Downloading template %s into %s', url, filename)
+
+    spec_content = get_url_content(url, caching=True)
+
+    # Check to see if no content was returned, indicating we have the latest version already
```

Comment:
This comment hints that there could be a function `should_perform_update` that returns true if the cache is old, false otherwise

Reply:
Good call, I separated the update logic into a separate method `url_has_newer_version`.